### PR TITLE
Include coverage from `unit`  tests

### PR DESCRIPTION
**Pull request by MoonlightSentinel**
Ensures that the `unit` tests are run for coverage builds and that the resulting coverage date is included in the report.

This required some changes s.t. coverage files are emitted to the right directory + name. See the individual commits for the actual changes + explaination.

CC @ljmf00

### CODE REVIEW

**Review by ljmf00 — COMMENTED** [review]
I still can't understand how coverage is merged. `unit_tests` target generates lst files inside `test` folder but `./src/run.d -u` don't, although, with this change, this might override those files in that situation? Can you give a rationale for why `--DRT` options doesn't work?

File: test/tools/unit_test_runner.d [hunk 1]
```diff
@@ -335,7 +339,9 @@ int main(string[] args)
         return 1;
     }
 
-    const test = execute(outputPath);
+    // Run in the project root directory s.t. potential coverage files are emitted there
+    // --DRT-covopt="merge:1 srcpath:<sourcePath> dstpath: <sourcePath> didn't work
```
[the remark this answers is not in the document]
Comment (MoonlightSentinel) [reply test/tools/unit_test_runner.d]: Updated the comment to explain the issue

**Comment by MoonlightSentinel** [comment]
> I still can't understand how coverage is merged. 

Good catch, the configuration option got lost alongside `dstpath`. It's now present again.

> `unit_tests` target generates lst files inside `test` folder but `./src/run.d -u` don't, 

The shortcut for `-u`  didn't override the environment + CWD as done for `unit_tests`. Fixed by the first commit.

> Can you give a rationale for why `--DRT` options doesn't work?

Pointing `dstpath` to the root directory didn't cause the the lst files to be emitted there. Don't know why and don't have the time to investigate at this point (especially given that changing the CWD works as well)

**Review by ljmf00 — APPROVED** [review]

**Comment by ljmf00** [comment]
> > I still can't understand how coverage is merged.
> 
> Good catch, the configuration option got lost alongside `dstpath`. It's now present again.

Cool. Just for context, for example, for the rest of the tests (`runnable`, `fail_compilation`, ...), how is the coverage merged in those situations?

> > Can you give a rationale for why `--DRT` options doesn't work?
> 
> Pointing `dstpath` to the root directory didn't cause the the lst files to be emitted there. Don't know why and don't have the time to investigate at this point (especially given that changing the CWD works as well)

Ah right, no worry. Maybe some weird logic on `rt/cover.d` in the runtime. I never used `dstpath`.

**Comment by ljmf00** [comment]
`macOS 10.15 x64, DMD (bootstrap)` failed. I restarted.

**Comment by MoonlightSentinel** [comment]
> Cool. Just for context, for example, for the rest of the tests (`runnable`, `fail_compilation`, ...), how is the coverage merged in those situations?

Those tests execute the dmd built by `build.d` with `ENABLE_COVERAGE=1`. The generated binary calls the corresponding functions from druntime on startup:

https://github.com/dlang/dmd/blob/7a5a76f2ce4f3f9cd965f1a58d037bf372c1b27a/src/dmd/mars.d#L977-L997

I've modified this PR to build the `unit` executable in the same way, s.t. the coverage files are consistently generated even when it is run independent of `unit_test_runner` (e.g. for debugging)

**Comment by ljmf00** [comment]
> > Cool. Just for context, for example, for the rest of the tests (`runnable`, `fail_compilation`, ...), how is the coverage merged in those situations?
> 
> Those tests execute the dmd built by `build.d` with `ENABLE_COVERAGE=1`. The generated binary calls the corresponding functions from druntime on startup:
> 
> https://github.com/dlang/dmd/blob/7a5a76f2ce4f3f9cd965f1a58d037bf372c1b27a/src/dmd/mars.d#L977-L997

Right. Makes sense. Thanks for explaining :)

> I've modified this PR to build the `unit` executable in the same way, s.t. the coverage files are consistently generated even when it is run independent of `unit_test_runner` (e.g. for debugging)

Ok, nice. I tested with https://github.com/dlang/dmd/pull/13625 and seems working correctly. Is this good to go @MoonlightSentinel ?

File: test/tools/unit_test_runner.d [hunk 2]
```diff
@@ -215,6 +215,18 @@ void writeRunnerFile(Range)(Range moduleNames, string path, string filter)
 
             return result;
         }
+
+        version (D_Coverage)
+        shared static this()
+        {
+            import core.runtime;
+
+            static immutable sourcePath = `%s`;
```
Comment (MoonlightSentinel) [inline test/tools/unit_test_runner.d:224]: Small update: The path is now set by `unit_test_runner` s.t. it also works when `RESULTS_DIR` is located outside of `test`.

Comment (ljmf00) [reply test/tools/unit_test_runner.d:224]: Ah, got it, never mind. Probably use `"` and escape `\"`.

Comment (MoonlightSentinel) [reply test/tools/unit_test_runner.d:224]: `%s` is replaced with `projectRootDir`. The string literal explicitly uses ``` s.t. it doesn't need to deal with the other escape sequences (e,g, `\` on windows)

**Review by ljmf00 — APPROVED** [review]
Looks good.

File: test/tools/unit_test_runner.d (hunk 2 above)
Comment (MoonlightSentinel) [reply test/tools/unit_test_runner.d:224]: FYI: This also supports file paths with spaces, even though the `unit` tests cannot be built in that case due to this [cmdfile bug](https://issues.dlang.org/show_bug.cgi?id=22773)

**Pull request merged**